helix-term/src/commands.rs

```diff
@@ -1308,6 +1309,8 @@ pub fn goto_mode(cx: &mut Context) {
                 'y' => goto_type_definition(cx),
                 'r' => goto_reference(cx),
                 'i' => goto_implementation(cx),
+                'n' => goto_next_diag(cx),
```

Comment:
I would like to see `[d` for previous and `]d` for next. Also `[D` for first and `]D` for last diagnostic in the buffer. You can implement `[` and `]` as `on_next_key` commands, similar to how `goto_mode` works.

Reply:
I was thinking `[e` since user is more familiar with "error" than "diagnostic" probably.

Reply:
What should happen to the currently bound `command::expand_selection` to `[`?

Reply:
> I was thinking `[e` since user is more familiar with "error" than "diagnostic" probably.

I think `d` for diagnostic makes more sense as this doesn't only apply to errors but also other diagnostics.